Gems/PhysX/Code/Source/Scene/PhysXScene.cpp

```diff
@@ -1287,4 +1307,46 @@ namespace PhysX
         }
     }
 
-}
+    void PhysXScene::FlushTransformSync()
+    {
+        AZ_PROFILE_SCOPE(Physics, "PhysX::FlushTransformSync");
+
+        m_queuedActiveBodyIndices.Apply(
+            [this](AzPhysics::SimulatedBodyIndex bodyIndex)
+            {
+                if (bodyIndex < m_simulatedBodies.size() && m_simulatedBodies[bodyIndex].second)
+                {
+                    m_simulatedBodies[bodyIndex].second->SyncTransform(m_accumulatedDeltaTime);
+                }
+            });
```

Comment:
This loop would be much faster if what's stored in the queue is `m_simulatedBodies[bodyIndex].second` directly instead of `AzPhysics::SimulatedBodyIndex`, that way when calling `Apply` it could just do:

`simulatedBody->SyncTransform(m_accumulatedDeltaTime);`

It removes the branching called for each element in the queue. Any reason not to store the pointer to the rigid body directly?

Reply:
That's the point of all these SimulatedBodyHandles (which is just scene name CRC + body Index) in the first place - we don't end up with dangling pointers which could happen if we stored `SimulatedBody*` directly.

Hence we do `if (bodyIndex < m_simulatedBodies.size() && m_simulatedBodies[bodyIndex].second)` check for each bodyIndex first